### DISCUSSION TITLE
Can't seem to make react-router to work with Vercel

### DISCUSSION THREAD

**Discussion by Hristo-Spasov** [vercel/community#6784] [Help] [answered]
### Summary

I recently deployed my Single Page Application (SPA) to Vercel, but I can't get the routing to work properly. I'm using React Router to manage the routing.

I've tried multiple solutions found online, including creating a vercel.json file and attempting various configurations. I also tried moving the vercel.json file from the root of the project to the public folder, but that didn't help.

My root page loads fine, but when I navigate to a different page, I receive a 404 error.

My route setup

```
const router = createBrowserRouter([
  {
    path: "/",
    element: (
      <QueryClientProvider client={queryClient}>
        <FetchProvider>
          <FlightProvider>
            <Root />
          </FlightProvider>
        </FetchProvider>
      </QueryClientProvider>
    ),
    errorElement: <ErrorPage />,
    children: [
      {
        index: true,
        element: <Home />,
      },
    ],
  },
  {
    path: "under-construction",
    element: <UnderConstruction />,
  },
]);

```
My navbar code

```
        <div className={style.nav_container}>
          <NavLink
            to="/"
            className={({ isActive, isPending }) =>
              isPending ? "pending" : isActive ? "active" : ""
            }
          >
            Home
          </NavLink>
          <NavLink
            to="under-construction"
            className={({ isActive, isPending }) =>
              isPending ? "pending" : isActive ? "active" : ""
            }
          >
            Tickets
          </NavLink>
```

vercel.json currently

`{   "rewrites": [{ "source": "/(.*)", "destination": "/" }] }`

package.json since I saw some people suggest to put "homepage"

```
{ 
"version": "0.0.0",
  "type": "module",
  "homepage": ".",
}
```

### Steps to Reproduce

![Screenshot 2024-05-09 222244](https://github.com/vercel/community/assets/119752462/127e4669-c7ca-4385-bb9a-d866b85c6bda)

**Comment by sahilatahar** [comment]
I think there is a potential problem with one of the routes. In the second route definition for "Under Construction" (`"/under-construction"`), the path is specified without a leading slash. 

### Example Correction:

```javascript
{
  path: "/under-construction",
  element: <UnderConstruction />,
}
```

**Comment by Hristo-Spasov** [comment, marked as answer]
It was cache related problem it's fixed now. Vercel was not recognising the vercel.json file.

**Reply by sahilatahar** [reply]
Glad to hear the issue is resolved!
